volume/repository_test.go

```diff
@@ -1086,4 +1093,131 @@ var _ = Describe("Repository", func() {
+				w.WriteHeader(serverResponseCode)
+				serverCalled = true
+
+				var err error
+				serverReadBytes, err = ioutil.ReadAll(r.Body)
+				Expect(err).ToNot(HaveOccurred())
+			}))
+			streamErr = repository.StreamP2pOut(context.Background(), "some-handle", filepath.Base(tempFile.Name()), volume.GzipEncoding, server.URL)
+		})
+
+		Context("when lookup volume fails", func() {
+			BeforeEach(func() {
+				fakeFilesystem.LookupVolumeReturns(nil, true, errors.New("some-error"))
+			})
+			It("should fail", func() {
+				Expect(streamErr).To(HaveOccurred())
+				Expect(streamErr).To(Equal(errors.New("some-error")))
+			})
+			It("should not http request", func() {
+				Expect(serverCalled).To(BeFalse())
+			})
+		})
+
+		Context("when volume not found", func() {
+			BeforeEach(func() {
+				fakeFilesystem.LookupVolumeReturns(nil, false, nil)
+			})
+			It("should fail", func() {
+				Expect(streamErr).To(HaveOccurred())
+				Expect(streamErr).To(Equal(volume.ErrVolumeDoesNotExist))
+			})
+			It("should not http request", func() {
+				Expect(serverCalled).To(BeFalse())
+			})
+		})
+
+		Context("when volume found", func() {
+			var (
+				fakeVolume *volumefakes.FakeFilesystemLiveVolume
+			)
+			BeforeEach(func() {
+				tempFile.WriteString("hello-world")
+				tempFile.Close()
+
+				fakeVolume = new(volumefakes.FakeFilesystemLiveVolume)
+				fakeVolume.DataPathReturns(filepath.Dir(tempFile.Name()))
+				fakeFilesystem.LookupVolumeReturns(fakeVolume, true, nil)
+			})
+
+			Context("when load privileged fails", func() {
+				BeforeEach(func() {
+					fakeVolume.LoadPrivilegedReturns(false, errors.New("some-error"))
+				})
+				It("should fail", func() {
+					Expect(streamErr).To(HaveOccurred())
+					Expect(streamErr).To(Equal(errors.New("some-error")))
+				})
+				It("should not http request", func() {
+					Expect(serverCalled).To(BeFalse())
+				})
+			})
+
+			Context("when load privileged ok", func() {
+				BeforeEach(func() {
+					fakeVolume.LoadPrivilegedReturns(false, nil)
+				})
+
+				Context("remote returns error", func() {
+					BeforeEach(func() {
+						serverResponseCode = http.StatusInternalServerError
+					})
+					It("should fail", func() {
+						Expect(streamErr).To(HaveOccurred())
+						Expect(streamErr).To(Equal(fmt.Errorf("p2p streaming error %d", http.StatusInternalServerError)))
+					})
+					It("should http request", func() {
+						Expect(serverCalled).To(BeTrue())
+					})
+				})
+
+				Context("remote returns ok", func() {
+					BeforeEach(func() {
+						serverResponseCode = http.StatusNoContent
+					})
+					It("should fail", func() {
+						Expect(streamErr).ToNot(HaveOccurred())
+					})
+					It("should http request", func() {
+						Expect(serverCalled).To(BeTrue())
+					})
+					It("remote should receive bytes", func() {
+						b := new(bytes.Buffer)
+						tgzfs.Compress(b, filepath.Dir(tempFile.Name()), filepath.Base(tempFile.Name()))
+						Expect(serverReadBytes).To(Equal(b.Bytes()))
+					})
```

Comment:
This test failed for me.
```
[Fail] Repository StreamP2pOut when volume found when load privileged ok remote returns ok [It] remote should receive bytes
/Users/taylor/workspace/concourse/baggageclaim/volume/repository_test.go:1218

Ran 11 of 134 Specs in 0.118 seconds
FAIL! -- 10 Passed | 1 Failed | 0 Pending | 123 Skipped
--- FAIL: TestVolume (0.12s)
```

Reply:
Looks like this was a flake. I noticed that it passed in CI and ran it again on my machine and now it keeps passing 🤷

Reply:
Yeah, I noticed it's flaky due to byte arrays' capacities might be different though len are same. I have enhanced the test.